### PR TITLE
ui: add redux for state management

### PR DESCRIPTION
**Pull request by kprzerwa**
* added api client config available in actions

### CODE REVIEW

File: invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/TimelineFeed.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class TimelineFeed extends Component {
+
+  componentDidMount() {
+    const { fetchTimeline } = this.props;
+    fetchTimeline();
+  }
+
+  render() {
+    const { timeline, loading } = this.props;
+    console.log(timeline, "AAAAA");
```
Comment (NRodriguezcuellar) [inline invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/TimelineFeed.js:17]: log spotted!

**Review by NRodriguezcuellar — COMMENTED** [review]
Looks good to me :smile:

File: invenio_requests/assets/semantic-ui/js/invenio_requests/InvenioRequestsApp.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+InvenioRequestsApp.propTypes = {
+  api: PropTypes.object,
+  overriddenCmps: PropTypes.object,
+  request: PropTypes.object.isRequired,
+};
+
+InvenioRequestsApp.defaultProps = {
+  overriddenCmps: { },
```
Comment (ntarocco) [inline invenio_requests/assets/semantic-ui/js/invenio_requests/InvenioRequestsApp.js:45]: missing `api` default, given that it is not required.

File: invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/TimelineFeed.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        </Container>
+      </Overridable>
+      </Loader>
+    );
+  }
+}
+
+TimelineFeed.propTypes = {
+  fetchTimeline: PropTypes.func.isRequired,
+  timeline: PropTypes.object,
```
Comment (ntarocco) [inline invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/TimelineFeed.js:37]: Missing `loading` and `error`, plus error handling in the cmp?

**Review by ntarocco — APPROVED** [review]
LGTM! :) 👏🏼

File: invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/TimelineFeed.js (hunk 3 above)
Comment (kprzerwa) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/TimelineFeed.js:37]: I will add in the next PR, I lost it during the refactoring, thanks for the catch

File: invenio_requests/assets/semantic-ui/js/invenio_requests/InvenioRequestsApp.js (hunk 2 above)
Comment (kprzerwa) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/InvenioRequestsApp.js:45]: 👍 I will add in the next PR

**Pull request merged**